### PR TITLE
Sheep TF + Plush Pseudo-TF Mods

### PR DESCRIPTION
**Pull request by Modenn**
Sheep TF:
- Clovis: Progressively transforms PC into wooly sheep
- Clover: Gives Clovis, sex scenes, bad end (found in Owca)

Plush Pseudo-TF:
- Homespun Apple: Progressively transforms PC into a childlike plush
doll of any race (compatible with any added races), bad end for
overconsumption
- Unspun Apple: Purified by Rathazul, reverts plush transformations
- Doll enemy: Gives Homespun Apple, sex scenes (found in TownRuins)
- Doll enemy will be able to be recruited as a follower/lover/slave in a
later update

### CODE REVIEW

File: classes/classes/Items/Consumables/HairDye.as [hunk 1]
```diff
@@ -71,10 +77,13 @@ package classes.Items.Consumables
 		}
 		
 		private function dyeFur():void {
+			var furname:String = "fur";
+			if (player.hasFleece())
+				furname = "fleece";
```
Comment (Stadler76) [inline classes/classes/Items/Consumables/HairDye.as]: I'd use
```as3
var furName:String = player.hasFur() ? "fur" : "fleece";
```
for this

File: classes/classes/Player.as [hunk 2]
```diff
@@ -1762,6 +1766,27 @@ use namespace kGAMECLASS;
 			return catCounter;
 		}
 		
+		//Sheep
+		public function sheepScore():Number
+		{
+			var sheepCounter:Number = 0;
+			if (earType == EARS_SHEEP)
+				sheepCounter += 1;
```
Comment (Stadler76) [inline classes/classes/Player.as]: `sheepCounter += 1;`? The increment operator exists: `sheepCounter++;`

File: includes/appearanceDefs.as [hunk 3]
```diff
@@ -106,6 +111,8 @@ public static const HORNS_ANTLERS:int
 public static const HORNS_GOAT:int                                                  =   6;
 public static const HORNS_UNICORN:int                                               =   7;
 public static const HORNS_RHINO:int                                                 =   8;
+public static const HORNS_SHEEP:int                                                 =   9;
+public static const HORNS_RAM:int        	                                        =  10;
```
Comment (Stadler76) [inline includes/appearanceDefs.as]: You're mixing tabs and spaces for alignment here. I'd suggest you the fix this.

File: includes/appearanceDefs.as [hunk 4]
```diff
@@ -160,6 +167,7 @@ public static const TAIL_TYPE_RHINO:int
 public static const TAIL_TYPE_ECHIDNA:int                                           =  23;
 public static const TAIL_TYPE_DEER:int                                              =  24;
 public static const TAIL_TYPE_SALAMANDER:int                                        =  25;
+public static const TAIL_TYPE_SHEEP:int                                        		=  26;
```
Comment (Stadler76) [inline includes/appearanceDefs.as]: Same as above: Tabs and spaces mixed

File: includes/appearanceDefs.as [hunk 5]
```diff
@@ -202,6 +210,7 @@ public static const LOWER_BODY_TYPE_CLOVEN_HOOFED:int
 public static const LOWER_BODY_TYPE_ECHIDNA:int                                     =  23;
 public static const LOWER_BODY_TYPE_DEERTAUR:int                                    =  24; // DEPRECATED, use LOWER_BODY_TYPE_CLOVEN_HOOFED and legCount=4
 public static const LOWER_BODY_TYPE_SALAMANDER:int                                  =  25;
+public static const LOWER_BODY_TYPE_SHEEP:int                                  		=  26;
```
Comment (Stadler76) [inline includes/appearanceDefs.as]: Moar of teh ebil tabs0rz.

File: classes/classes/Scenes/NPCs/DollScene.as [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+				player.orgasm();
+			}
+			else {
+				outputText(" The doll's aura becomes a deep pink, which turns to an angry red as it realizes you haven't even got something to fuck it with.");
+			}
+			combat.cleanupAfterCombat();
+		}
+
+		// Win
+		public function victory():void {
```
Comment (Stadler76) [inline classes/classes/Scenes/NPCs/DollScene.as]: It is recommended, that in function definitions the opening brace of the function goes into a new line.

Well, there are a few exemptionp, but not here 😉

File: classes/classes/Appearance.as [hunk 7]
```diff
@@ -2209,6 +2209,7 @@
 					[SKIN_TYPE_UNDEFINED, "undefined flesh"],
 					[SKIN_TYPE_DRAGON_SCALES, "scales"],
 					[SKIN_TYPE_FISH_SCALES, "scales"],
+					[SKIN_TYPE_FLEECE, "fleece"]
```
Comment (Stadler76) [inline classes/classes/Appearance.as]: I'd suggest, that you leave a trailing comma in the last line here. So future diffs are smaller. Its not mandatory, but it's common practice.

Same in the changes below.

File: classes/classes/Creature.as [hunk 8]
```diff
@@ -2462,7 +2462,17 @@ package classes
...
+		
+		public function hasFleece():Boolean
+		{
+			return skinType == SKIN_TYPE_FLEECE;
+		}
+		
+		public function hasFurOrFleece():Boolean
+		{
+			return hasFur() || hasFleece();
+		}
```
Comment (Stadler76) [inline classes/classes/Creature.as]: I'd suggest to refactor that function to be named isFurry() so we don't need to rename it every time, when we add more skintypes. `hasFurOrFleeceOrBristlesOrWhatever` would sound kinda awkward anyway 😕

File: classes/classes/Player.as (hunk 2 above)
Comment (Modenn) [inline classes/classes/Player.as]: this is personal preference, not an error in code

File: classes/classes/Scenes/NPCs/DollScene.as (hunk 6 above)
Comment (Modenn) [inline classes/classes/Scenes/NPCs/DollScene.as]: again, this is personal preference, not an error in code

File: classes/classes/Creature.as (hunk 8 above)
Comment (JustGregory) [inline classes/classes/Creature.as]: Why not name it ```hasSkinCover``` then...

Comment (JustGregory) [inline classes/classes/Creature.as]: Why not name it ```hasSkinCover``` then...

**Pull request closed without merging**